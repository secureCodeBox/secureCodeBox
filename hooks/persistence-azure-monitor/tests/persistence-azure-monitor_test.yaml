# SPDX-FileCopyrightText: the secureCodeBox authors
#
# SPDX-License-Identifier: Apache-2.0

suite: Full Snapshot

tests:
  - it: matches the snapshot
    chart:
      version: 0.0.0
      appVersion: 0.0.0  
    set:
      ## Azure Monitor specific configuration
      monitor.authentication.apiKeySecret: {name: foo, key: bar}
      ## Generic Configuration
      imagePullSecrets: [{name: foo}]
      hook:
        labels: {foo: bar}
        annotations: {foo: bar}
        priority: 100
        ttlSecondsAfterFinished: 100
        tolerations:
          [{key: foo, operator: Exists, value: "value1", effect: NoSchedule}]
        affinity:
          nodeAffinity:
            requiredDuringSchedulingIgnoredDuringExecution:
              nodeSelectorTerms:
                - matchExpressions:
                    - {key: foo, operator: In, values: [bar, baz]}
        resources: {requests: {cpu: "100m", memory: "100Mi"}}
        extraVolumes: [{name: foo, emptyDir: {}}]
        extraVolumeMounts: [{name: foo, mountPath: /foo}]
        env: [{name: foo, value: bar}]
    asserts:
      - matchSnapshot: {}
