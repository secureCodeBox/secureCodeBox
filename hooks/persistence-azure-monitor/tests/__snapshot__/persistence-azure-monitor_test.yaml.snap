matches the snapshot:
  1: |
    raw: |2
      Azure Monitor PersistenceProvider deployed.
  2: |
    apiVersion: execution.securecodebox.io/v1
    kind: ScanCompletionHook
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: persistence-azure-monitor
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: persistence-azure-monitor-0.0.0
        type: Structured
      name: RELEASE-NAME-persistence-azure-monitor
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                  - key: foo
                    operator: In
                    values:
                      - bar
                      - baz
      env:
        - name: MONITOR_LOGTYPE_PREFIX
          value: SCB
        - name: MONITOR_WORKSPACE_ID
          valueFrom:
            secretKeyRef:
              key: workspace
              name: map[key:bar name:foo]
        - name: MONITOR_SHARED_KEY
          valueFrom:
            secretKeyRef:
              key: sharedkey
              name: map[key:bar name:foo]
        - name: foo
          value: bar
      image: docker.io/securecodebox/hook-persistence-azure-monitor:0.0.0
      imagePullSecrets:
        - name: foo
      priority: 100
      resources:
        requests:
          cpu: 100m
          memory: 100Mi
      tolerations:
        - effect: NoSchedule
          key: foo
          operator: Exists
          value: value1
      ttlSecondsAfterFinished: 100
      type: ReadOnly
      volumeMounts:
        - mountPath: /foo
          name: foo
      volumes:
        - emptyDir: {}
          name: foo
